.github/workflows/shared-build.yml

```diff
@@ -11,9 +11,6 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - name: 🛑 Cancel Previous Runs
-        uses: styfle/cancel-workflow-action@0.11.0
```

Comment:
We shouldn't need this and can use the built in [`concurrency`](https://docs.github.com/en/actions/using-jobs/using-concurrency) setting, which is recommended on the [README](https://github.com/styfle/cancel-workflow-action#readme) for this action these days as well.

This also allows us better control over cancellation behavior per workflow.  We do not want to cancel runs on `dev` for example since they may come in repeatedly and we don't want a bunch of cancellations queueing up which may obscure the commit which broke something:

<img width="878" alt="Screenshot 2023-08-28 at 10 09 11 AM" src="https://github.com/remix-run/remix/assets/1609022/ad16204e-aba9-4023-ba47-bd3e1847fb0f">


So we only use the concurrency setting on the PR workflows, not the `test.yml` workflow

Reply:
We use this `cancel-workflow-action` in a bunch of other places, if we see this approach works without issue we can move those to use `concurrency` in a separate PR